base/linalg/rowvector.jl

```diff
@@ -0,0 +1,183 @@
+immutable RowVector{T,V<:AbstractVector} <: AbstractMatrix{T}
```

Comment:
@alanedelman and I both tend to think that `Transpose{T,N,AbstractArray{T,N}} <: AbstractArray{T,N}`, so that a `RowVector` (`AbstractRowVector`? = `Transpose{T,1,AbstractArray{T,1}}`) would be a subtype of `AbstractVector`, i.e. a "1d" array object.